python/python/bystro/ancestry/gmm_ancestry.py

```
@@ -0,0 +1,436 @@
+        if prior_options is None:
+            prior_options = {}
+
+        self.training_options = self._fill_training_options(training_options)
+        self.prior_options = self._fill_prior_options(prior_options)
+
+    def _fill_training_options(self, training_options):
+        """
+        This sets the default parameters for stochastic gradient descent,
+        our inference strategy for the model.
+
+        Parameters
+        ----------
+        training_options : dict
+            The original options set by the user passed as a dictionary
+
+        Options
+        -------
+        n_iterations : int, default=3000
+            Number of iterations to train using stochastic gradient descent
+
+        learning_rate : float, default=1e-4
+            Learning rate of gradient descent
+
+        batch_size : int, default=None
+            The number of observations to use at each iteration. If none
+            corresponds to batch learning
+        """
+        default_options = {
+            "n_iterations": 3000,
+            "learning_rate": 1e-3,
+            "batch_size": 100,
+            "momentum": 0.9,
+        }
+        tops = {**default_options, **training_options}
+
+        default_keys = set(default_options.keys())
+        final_keys = set(tops.keys())
+
+        expected_but_missing_keys = default_keys - final_keys
+        unexpected_but_present_keys = final_keys - default_keys
+        if expected_but_missing_keys:
+            raise ValueError("Missing keys")
+        if unexpected_but_present_keys:
+            raise ValueError("Extra keys")
+        return tops
+
+    def _fill_prior_options(self, prior_options):
+        """
+        Fills in options for prior parameters on latent space
+
+        Paramters
+        ---------
+        new_dict : dictionary
+            The prior parameters used to specify the prior
+        """
+        default_dict = {"mu_l2": 1.0}
+        new_dict = {**default_dict, **prior_options}
```

Do we need to raise a value error for expected priors?

---

I'll come up with better priors later. Immediate task will be creating a more meaningful unit test and making sure it fits well.